wireapp/wire-android
# fix: German translations for adding, removing and leaving groups [AR-3190][AR-3192]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3190" title="AR-3190" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-3190</a>  Removing, adding a user, and self user leaving group have wrong translations in german
  </td></table>
  <br />
 




----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Translations were broken for German locale. They were not handling correctly the cases where self user was removing others, or being removed by other. Same for being added to groups. Therefore, a deeper refactor was needed to be written mapping each case to the full sentence as the German grammar wasn't allowing partial translations of smaller chunks of the phrase.


#### How to Test

Add, remove, users to groups, with different accounts, and check the right strings are shown in English and German.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
